test/memory.jl

```diff
@@ -10,58 +10,81 @@ let
 end
 =#
 
-# pointer-based
-src = 42
+let
```

Comment:
Can we just turn these `let`s into `@testset`s, with the testset name being the same as the first comment (like `@testset "Pointer-based"`)?